packages/framer-motion/src/value/scroll/use-element-scroll.ts

```diff
@@ -6,9 +6,12 @@ import { useScroll } from "../use-scroll"
  * @deprecated useElementScroll is deprecated. Convert to useScroll({ container: ref })
  */
 export function useElementScroll(ref: RefObject<HTMLElement>) {
-    warnOnce(
-        false,
-        "useElementScroll is deprecated. Convert to useScroll({ container: ref })."
-    )
+    if (process.env.NODE_ENV === "development") {
```

Comment:
What is the reason to sometimes use `process.env.NODE_ENV === "development"` and sometimes `process.env.NODE_ENV !== "production"`?

Reply:
There's some tests in there that check whether `console.warn` was called and this is to catch `NODE_ENV === "test"`